modules/Documents/SaveFile.php

```diff
@@ -58,4 +58,36 @@
 			echo 'file_available';
 		}
 }
+
+if(isset($_REQUEST['act']) && $_REQUEST['act'] == 'massDldCnt')
+{
+	$all_files = $_REQUEST['file_id'];
+	$zipfilename = "cache/Documents.zip";
```

Comment:
add the userid here

cache/Documents{userid}.zip

that way if two or more users are downloading they don't overwrite each other.